dp/generate_climos.py

```diff
@@ -383,44 +385,45 @@ def convert_longitude_range(climo_data):
-        pr_units = Unit.from_udunits_str(pr_variable.units)
-        if pr_units in [Unit('kg / m**2 / s'), Unit('mm / s')]:
-            logger.info("Converting 'pr' variable to units mm/day")
+    attributes = {}  # will contain updates, if any, to pr variable attributes
+    flux_vars = [var for var in ['pr', 'prsn'] if var in input_file.dependent_varnames()]
+
+    for flux_var in flux_vars:
+        flux_variable = input_file.variables[flux_var]
+        units = Unit.from_udunits_str(flux_variable.units)
+        if units in [Unit('kg / m**2 / s'), Unit('mm / s')]:
+            logger.info("Converting {} variable to units mm/day".format(flux_var))
             # Update units attribute
-            pr_attributes['units'] = (pr_units * Unit('s / day')).to_udunits_str()
+            attributes['units'] = (units * Unit('s / day')).to_udunits_str()
             # Multiply values by 86400 to convert from mm/s to mm/day
             seconds_per_day = 86400
-            if hasattr(pr_variable, 'scale_factor') or hasattr(pr_variable, 'add_offset'):
+            if hasattr(flux_variable, 'scale_factor') or hasattr(flux_variable, 'add_offset'):
                 # This is a packed file; need only modify packing parameters
                 try:
-                    pr_attributes['scale_factor'] = seconds_per_day * pr_variable.scale_factor
+                    attributes['scale_factor'] = seconds_per_day * flux_variable.scale_factor
                 except AttributeError:
-                    pr_attributes['scale_factor'] = seconds_per_day * 1.0  # default value 1.0 for missing scale factor
+                    attributes['scale_factor'] = seconds_per_day * 1.0  # default value 1.0 for missing scale factor
                 try:
-                    pr_attributes['add_offset'] = seconds_per_day * pr_variable.add_offset
+                    attributes['add_offset'] = seconds_per_day * flux_variable.add_offset
                 except AttributeError:
-                    pr_attributes['add_offset'] = 0.0  # default value 0.0 for missing offset
+                    attributes['add_offset'] = 0.0  # default value 0.0 for missing offset
             else:
                 # This is not a packed file; modify the values proper
                 # Extract variable
-                pr_only = cdo.select('name=pr', input=climo_data)
+                var_only = cdo.select('name={}'.format(flux_var), input=climo_data)
                 # Multiply values by 86400 to convert from mm/s to mm/day
-                pr_only = cdo.mulc(str(seconds_per_day), input=pr_only)
+                var_only = cdo.mulc(str(seconds_per_day), input=var_only)
                 # Replace pr in all-variables file
-                climo_data = cdo.replace(input=[climo_data, pr_only])
+                climo_data = cdo.replace(input=[climo_data, var_only])
 
     # Update pr variable metadata as necessary to reflect changes madde
     with CFDataset(climo_data, mode='r+') as cf:
```

Comment:
I think you're only updating attributes for one of the variables in cases where there are two. This `with` should be inside the `for`, and possibly `attributes` needs to be reset to empty after each variable.

Reply:
Ah you're right, my mistake there.